### PR TITLE
Feature/18295 ps report data service refactor to sql server

### PR DESCRIPTION
**Pull request by jon-shipley**

### CODE REVIEW

File: admin/app.js [hunk 1]
```diff
@@ -29,9 +29,8 @@ const azure = require('./azure')
  * use LogDNA transport for winston if configuration setting available
  */
 const winston = require('winston')
-let logdna
 if (config.Logging.LogDna.key) {
-  logdna = require('logdna')
+  require('logdna')
```
Comment (GuyHarwood) [inline admin/app.js]: 👍

File: admin/services/certificate-store.service.js [hunk 2]
```diff
@@ -2,24 +2,18 @@ const azureBlobDataService = require('./data-access/azure-blob.data.service')
 const config = require('../config')
 
 let ncaPublicKey, mtcPrivateKey
-let callCount = 0
```
Comment (GuyHarwood) [inline admin/services/certificate-store.service.js]: thanks for tidying this up 👍

File: admin/services/date.service.js [hunk 3]
```diff
@@ -35,6 +36,7 @@ const dateService = {
 
   checkAndFormat: function (date, format) {
     if (!(date instanceof Date || date instanceof moment)) {
+      winston.warn(`Date parameter is not a Date or Moment object: ${date}`)
       return ''
```
Comment (GuyHarwood) [inline admin/services/date.service.js]: i wonder if we should be throwing an error here?  or at least returning false 🤔

Comment (jon-shipley) [reply admin/services/date.service.js]: Yes, I've considered that, but if we did then we wouldn't be able to directly use the output of dateService in the UI, we would have to catch errors and provide defaults.

**Pull request merged**